tests/Doctrine/Tests/ODM/PHPCR/Mapping/ClassMetadataTest.php

```diff
@@ -324,6 +324,91 @@ public function testClassMetadataInstanceSerializationTranslationProperties($cm)
         $this->assertTrue(in_array('translatedField', $cm->translatableFields));
         $this->assertEquals('locale', $cm->localeMapping);
     }
+
+    /**
+     * It should throw an exception if given a child class FQN when the
+     * metadata is for a leaf.
+     *
+     * @expectedException \Doctrine\ODM\PHPCR\Exception\OutOfBoundsException
+     * @expectedExceptionMessage has been mapped as a leaf
+     */
+    public function testAssertValidChildClassesIsLeaf()
```

Comment:
Much easier to test the classmetadata rather than the UOW for all these code paths.